base/range.jl

```diff
@@ -1130,9 +1130,7 @@ AbstractUnitRange{T}(r::UnitRange) where {T} = UnitRange{T}(r)
 AbstractUnitRange{T}(r::OneTo) where {T} = OneTo{T}(r)
 
 OrdinalRange{T1, T2}(r::StepRange) where {T1, T2<: Integer} = StepRange{T1, T2}(r)
-OrdinalRange{T1, T2}(r::AbstractUnitRange{T1}) where {T1, T2<:Integer} = r
-OrdinalRange{T1, T2}(r::UnitRange) where {T1, T2<:Integer} = UnitRange{T1}(r)
-OrdinalRange{T1, T2}(r::OneTo) where {T1, T2<:Integer} = OneTo{T1}(r)
```

Comment:
I think these methods are still needed:

```julia
julia> OrdinalRange{Int, Int8}(1:10)
1:10 # master
MethodError # current PR
```

Reply:
What should this produce? I don't think that the behavior on master is correct.

```julia
julia> OrdinalRange{Int, Int8}(1:10)
1:10

julia> OrdinalRange{Int, Int8}(1:10) |> step |> typeof
Int64
```

However the docstring of `OrdinalRange` states that `OrdinalRange{T, S}` has spacings of type `S`. The present behavior on master is not consistent with the docstring.

Reply:
Okay, that makes sense. Given that the previous 4 methods are newly added in Julia 1.6 only(#37829), I believe it's totally fine to revert that with `OrdinalRange{T, T}(r::AbstractUnitRange)` method.